### PR TITLE
[cosmos] Misc code cleanup

### PR DESCRIPTION
**Pull request by jayantk**
Clean up a few miscellaneous issues from earlier PRs. Main changes are:

* Use PythDataSource consistently across initialization messages and configs
* Use Coin in the config and initialization message instead of separate fee/denom parameters

I suggest reading `msg.rs` and `state.rs` first to get the main changes before looking at `contract.rs` (which is mostly mechanical)..

### CODE REVIEW

File: cosmwasm/contracts/pyth/src/contract.rs [hunk 1]
```diff
@@ -84,20 +83,13 @@ pub fn instantiate(
...
-        governance_source:          PythDataSource {
-            emitter:            msg.governance_emitter,
-            pyth_emitter_chain: msg.governance_emitter_chain,
-        },
+        governance_source:          msg.governance_source.clone(),
         governance_source_index:    msg.governance_source_index,
         governance_sequence_number: msg.governance_sequence_number,
         valid_time_period:          Duration::from_secs(msg.valid_time_period_secs as u64),
         fee:                        msg.fee,
-        fee_denom:                  msg.fee_denom,
```
Comment (jayantk) [inline cosmwasm/contracts/pyth/src/contract.rs:100]: `fee` is a Coin now so it has the denom already

File: cosmwasm/contracts/pyth/src/contract.rs [hunk 2]
```diff
@@ -130,23 +122,34 @@ fn update_price_feeds(
     mut deps: DepsMut,
     env: Env,
     info: MessageInfo,
-    data: &Binary,
+    data: &[Binary],
```
Comment (jayantk) [inline cosmwasm/contracts/pyth/src/contract.rs:125]: Pass in an array of price updates to update_price_feeds for consistency with the EVM implementation

File: cosmwasm/contracts/pyth/src/contract.rs [hunk 3]
```diff
@@ -183,49 +186,7 @@ fn execute_governance_instruction(
...
-                            format!("{}", parsed_claim_vaa.emitter_chain),
-                        )
-                        .add_attribute(
-                            "new_governance_sequence_number",
-                            format!("{}", parsed_claim_vaa.sequence),
-                        )
-                }
-                _ => Err(PythContractError::InvalidGovernancePayload)?,
-            }
+            transfer_governance(&mut updated_config, &state, &parsed_claim_vaa)?
```
Comment (jayantk) [inline cosmwasm/contracts/pyth/src/contract.rs:189]: moved to helper function (per previous PR comment)

File: cosmwasm/contracts/pyth/src/contract.rs [hunk 4]
```diff
@@ -271,11 +232,63 @@ fn execute_governance_instruction(
...
+    }
+}
+
 /// Check that `vaa` is from a valid data source (and hence is a legitimate price update message).
 fn verify_vaa_from_data_source(state: &ConfigInfo, vaa: &ParsedVAA) -> StdResult<()> {
     let vaa_data_source = PythDataSource {
-        emitter:            vaa.emitter_address.clone().into(),
-        pyth_emitter_chain: vaa.emitter_chain,
+        emitter:  vaa.emitter_address.clone().into(),
+        chain_id: vaa.emitter_chain,
```
Comment (jayantk) [inline cosmwasm/contracts/pyth/src/contract.rs:291]: (I renamed this field in data source)

File: cosmwasm/contracts/pyth/src/governance.rs [hunk 5]
```diff
@@ -1,11 +1,5 @@
 use {
-    crate::{
-        governance::GovernanceAction::{
-            RequestGovernanceDataSourceTransfer,
-            SetValidPeriod,
-        },
-        state::PythDataSource,
-    },
```
Comment (jayantk) [inline cosmwasm/contracts/pyth/src/governance.rs:8]: fix same file imports (per previous PR)

File: cosmwasm/contracts/pyth/src/contract.rs [hunk 6]
```diff
@@ -384,13 +396,13 @@ fn update_price_feed_if_new(
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
     match msg {
-        QueryMsg::PriceFeed { id } => to_binary(&query_price_feed(deps, env, id.as_ref())?),
-        QueryMsg::GetUpdateFee { vaas } => to_binary(&get_update_fee(deps, &vaas)?),
-        QueryMsg::GetValidTimePeriod => to_binary(&get_valid_time_period(deps)?),
+        QueryMsg::PriceFeed { id } => to_binary(&query_price_feed(&deps, env, id.as_ref())?),
+        QueryMsg::GetUpdateFee { vaas } => to_binary(&get_update_fee(&deps, &vaas)?),
+        QueryMsg::GetValidTimePeriod => to_binary(&get_valid_time_period(&deps)?),
```
Comment (jayantk) [inline cosmwasm/contracts/pyth/src/contract.rs:401]: changed signature to borrow (for consistency with the rest of the functions)

**Review by ali-bahjati — APPROVED** [review]
Awesome!

**Pull request merged**